dolphin/drivers/api.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+
+from oslo_log import log
+from oslo_utils import uuidutils
+
+from dolphin.drivers import manager
+
+LOG = log.getLogger(__name__)
+
+
+class API(object):
+    def __init__(self):
+        self.driver_manager = manager.DriverManager()
+
+    def register_storage(self, context, access_info):
+        """Discovery a storage system with access information."""
+        access_info['storage_id'] = six.text_type(uuidutils.generate_uuid())
+        driver = self.driver_manager.get_driver(context, **access_info)
+
+        storage = driver.register_storage(context, access_info)
```

Comment:
why need access_info in register_storage if you have already passed access_info to driver.__init__(**kwargs)

Reply:
Done, please check again.